### PR TITLE
Switch to v2 and implement crypto keys

### PR DESCRIPTION
**Pull request by setrofim**
- Switch to `v2` development, as, going forward, we're going to be implementing breaking changes from the v1 releases
- Implement Crypto Keys as defined by the latest draft of the CoRIM spec
- Update a deprecated `go-cose` algorithm name

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,9 +4,14 @@
 [![linters](https://github.com/veraison/corim/actions/workflows/linters.yml/badge.svg)](https://github.com/veraison/corim/actions/workflows/linters.yml)
 [![Go Reference](https://pkg.go.dev/badge/github.com/veraison/corim.svg)](https://pkg.go.dev/github.com/veraison/corim)
 
-The [`corim/corim`](corim) and [`corim/comid`](comid) packages provide a golang API for low-level manipulation of [Concise Reference Integrity Manifest (CoRIM)](https://datatracker.ietf.org/doc/draft-birkholz-rats-corim/) and Concise Module Identifier (CoMID) tags respectively.
+The [`corim/v2/corim`](corim) and [`corim/v2/comid`](comid) packages provide a golang API for low-level manipulation of [Concise Reference Integrity Manifest (CoRIM)](https://datatracker.ietf.org/doc/draft-birkholz-rats-corim/) and Concise Module Identifier (CoMID) tags respectively.
```
Comment (thomas-fossati) [inline README.md:7]:
```suggestion
The [`corim/v2/corim`](corim) and [`corim/v2/comid`](comid) packages provide a golang API for low-level manipulation of [Concise Reference Integrity Manifest (CoRIM)](https://datatracker.ietf.org/doc/draft-ietf-rats-corim/) and Concise Module Identifier (CoMID) tags respectively.
```

File: comid/cryptokey.go [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+
+	switch value.Type {
+	case PKIXBase64KeyType:
+		o.Value = TaggedPKIXBase64Key(value.Value)
+	case PKIXBase64CertType:
+		o.Value = TaggedPKIXBase64Cert(value.Value)
+	case PKIXBase64CertPathType:
+		o.Value = TaggedPKIXBase64CertPath(value.Value)
+	case COSEKeyType:
+		bytes, err := base64.StdEncoding.DecodeString(value.Value)
```
Comment (thomas-fossati) [inline comid/cryptokey.go:192]: hidden lint message: _"shadow of imported package 'bytes'"_

Comment (thomas-fossati) [reply comid/cryptokey.go:192]: > PS: We really should merge the "extra" lint target into the "normal" one as we've have done for other packages.

This is now https://github.com/veraison/corim/issues/91

**Review by thomas-fossati — APPROVED** [review]
LGTM!

File: README.md [hunk 3]
```diff
@@ -4,9 +4,14 @@
...
 
-The [`corim/corim`](corim) and [`corim/comid`](comid) packages provide a golang API for low-level manipulation of [Concise Reference Integrity Manifest (CoRIM)](https://datatracker.ietf.org/doc/draft-birkholz-rats-corim/) and Concise Module Identifier (CoMID) tags respectively.
+The [`corim/v2/corim`](corim) and [`corim/v2/comid`](comid) packages provide a golang API for low-level manipulation of [Concise Reference Integrity Manifest (CoRIM)](https://datatracker.ietf.org/doc/draft-ietf-rats-corim/) and Concise Module Identifier (CoMID) tags respectively.
 
-The [`corim/cocli`](cocli) package uses the API above (as well as the API from [`veraison/swid`](https://github.com/veraison/swid) package) to provide a user friendly command line interface for working with CoRIM, CoMID, CoSWID and CoTS.  Specifically it allows creating, signing, verifying, displaying, uploading, and more.  See [`cocli/README.md`](cocli/README.md) for further details.
+> [!NOTE]
+> These API are still in active development (as is the underlying CoRIM spec).
+> They are **subject to change** going forward, and should not be considered
```
Comment (yogeshbdeshpande) [inline README.md:11]: I think, the statement can reworded a bit in a positive way. The intention should be clear but I have an issue with the text. I would like to trim it down

Comment (yogeshbdeshpande) [inline README.md:11]:
```suggestion
> They are **subject to change** in future. 
```

File: README.md [hunk 4]
```diff
@@ -4,9 +4,14 @@
...
+> stable at this time.
```
Comment (yogeshbdeshpande) [inline README.md:12]:
```suggestion

```

**Review by yogeshbdeshpande — APPROVED** [review]
I have made some minor comments!

**Pull request merged**